### DISCUSSION TITLE
A question about @dependabot and resolving dependency updates

### DISCUSSION THREAD

**Discussion by doeg** [reactjs/react-modal#1042] [General]
Right now, there are [~20 PRs opened by dependabot](https://github.com/reactjs/react-modal/pulls) to update various dependencies.

Further, `npm audit` shows a handful of issues to be addressed. (I don't really love `npm audit` as a marker for vulnerability severity, but some projects do take it seriously.)

```
58 vulnerabilities (3 low, 14 moderate, 31 high, 10 critical)
```

In the past, I've personally preferred to do module updates myself to ensure that nothing breaks, since semver is not always trustworthy. :) That said, it's a ton of work to do it manually and I do see the appeal of @dependabot.

@diasbruno do you have thoughts on how you like to deal with @dependabot pull requests and module updates in general? 🙏 This is definitely something that I'd be happy to help with.

**Comment by diasbruno** [comment]
It would be nice to merge, but at this point - and as we already have plans to upgrade the toolchain, better update everything at once.

I've made progress using esbuild. Just need to put the examples back to work and compile the minified versions.

After the toolchain upgrade, we can just merge every time dependabot opens a PR.

**Comment by aniruddhaadak80** [comment]
Hi there! Taking a look at your issue regarding `A question about @dependabot and resolving dependency updates` in the context of `reactjs/react-modal`...

For `react-modal` integration specifically, this typically happens due to how the modal mounts into the portal DOM node vs standard React tree flow.
1. **App Element:** Ensure you've called `Modal.setAppElement('#root')` early in your app lifecycle so accessibility features (like `aria-hidden`) perform correctly without throwing errors.
2. **React 18/19 compatibility:** If encountering strict mode double mounts or `findDOMNode` warnings, verify that you are leveraging modern standard refs (`useRef`) and check your test environment (e.g. `jsdom`) where Portal mounting behaves slightly differently than standard browsers.
3. **Z-index and stacking:** Always check the stacking context of the `ReactModal__Overlay` class if the modal is hidden behind other components.

Best of luck, and hope this guides you to the fix!
